pkg/frontend/nvme_controller.go

```diff
@@ -98,6 +98,11 @@ func (s *Server) CreateNvmeController(_ context.Context, in *pb.CreateNvmeContro
 	if in.NvmeController.Spec == nil || in.NvmeController.Spec.SubsystemNameRef == "" {
 		return nil, status.Error(codes.InvalidArgument, "invalid input subsystem parameters")
 	}
+	// Validate that a resource name conforms to the restrictions outlined in AIP-122.
+	if err := resourcename.Validate(in.NvmeController.Spec.SubsystemNameRef); err != nil {
```

Comment:
I am thinking if we really need to check if we have a valid name for a subsystem here? It says nothing to us. Even a valid subsystem name can just lead to no entry. However, if we go to the db and such a subsystem exists within, it means the name is correct since it passed all the validation steps at creation. Also we have a single validation point in that case...

Reply:
I thought about this, but I feel it is better to check in order to give better explanation why DB is missing entry... might be helpful to users...

Reply:
Not sure if it would help to user. If it was not found -> it had not been created before. We are just adding cognitive load and burden of maintenance.
I believe we need to check only at create/update and search in db for other calls